includes/payment-methods/class-wc-stripe-payment-request.php

```diff
@@ -1544,14 +1483,30 @@ protected function build_display_items( $itemized_display_items = false ) {
 			];
 		}
 
-		return [
-			'displayItems' => $items,
-			'total'        => [
-				'label'   => $this->total_label,
-				'amount'  => max( 0, apply_filters( 'woocommerce_stripe_calculated_total', WC_Stripe_Helper::get_stripe_amount( $order_total ), $order_total, WC()->cart ) ),
-				'pending' => false,
-			],
+		// Order total
+		if ( version_compare( WC_VERSION, '3.2', '<' ) ) {
+			$items_total = wc_format_decimal( WC()->cart->cart_contents_total, WC()->cart->dp );
+			$order_total = wc_format_decimal( $items_total + $tax + $shipping, WC()->cart->dp );
+		} else {
+			// Getting the total amount from the cart automatically adds a shipping cost to it
+			// We need to remove it if the user hasn't picked a shipping address yet
+			$order_total = WC()->cart->get_total( false ) - $shipping_to_substract;
+		}
+
+		// Mandatory payment details.
+		$data['needs_payer_phone'] = 'required' == get_option( 'woocommerce_checkout_phone_field', 'required' );
```

Comment:
I get a warning here on not using strict comparison `===`.

Reply:
done 078e67a

Out of curiosity I assume you are using your editor's built-in linter? It would be nice to have a lint task like we have for php (`npm run lint:php`) or at least a `.eslintrc.js` so everyone can use it with their own editor.

Reply:
No, it's a phpcs warning (WordPress.PHP.StrictComparisons.LooseComparison). It probably doesn't show up on `lint:php` because it's a warning and not an error.

Reply:
I'm using VSCode [ikappas.phpcs](https://marketplace.visualstudio.com/items?itemName=ikappas.phpcs) extension to view these warnings.